core/server/services/mega/post-email-serializer.js

```diff
@@ -74,6 +77,22 @@ const createUnsubscribeUrl = (uuid, newsletterUuid) => {
     return unsubscribeUrl.href;
 };
 
+/**
+ * createPostSignupUrl
+ *
+ * Takes a post slug. Returns the url that should be used to signup from newsletter
+ *
+ * @param {string} slug post slug
+ */
+const createPostSignupUrl = (slug) => {
+    const siteUrl = urlUtils.getSiteUrl();
+    const signupUrl = new URL(siteUrl);
+    signupUrl.pathname = `${signupUrl.pathname}/${slug}/`.replace('//', '/');
```

Comment:
Do we not have a more authoritative way to generate the URL for a post?

Reply:
@matthanley Good pointer, I have updated this to use urlService now to fetch the post url the same way as we do it in serializer, including falling back to home url in case of email-only posts 👍